client_code/persistence.py

```diff
@@ -49,86 +49,103 @@ def __set__(self, instance, value):
+        self._args = args or []
+        self._kwargs = kwargs or {}
 
-def _getattr(self, key):
-    if self._delta and key in self._delta:
-        return self._delta[key]
+    def __set_name__(self, owner, name):
+        if self._linked_column is None:
+            self._linked_column = name
 
-    return dict(self._store).get(key, None)
+    def __get__(self, instance, objtype=None):
+        if instance is None:
+            return self
 
+        if self._obj is None:
+            self._obj = self._cls(
+                instance._store[self._linked_column], *self._args, **self._kwargs
+            )
 
-def _getitem(self, key):
-    return getattr(self, key)
+        return self._obj
 
+    def __set__(self, instance, value):
+        raise ValueError("Linked Class instance is already set and cannot be changed")
 
-def _set_value(self, key, value):
-    is_private = key.startswith("_")
-    is_descriptor = hasattr(self.__class__, key) and hasattr(
-        getattr(self.__class__, key), "__set__"
-    )
-    if is_private or is_descriptor:
-        object.__setattr__(self, key, value)
-    else:
-        self._delta[key] = value
 
+class PersistedClass:
+    @classmethod
+    def search(cls, *args, **kwargs):
+        rows = anvil.server.call(f"search_{cls.__name__.lower()}", *args, **kwargs)
+        return (cls.create(store=row) for row in rows)
 
-def _class_name(instance):
-    return "".join(
-        "_" + c.lower() if c.isupper() else c for c in instance.__class__.__name__
-    ).lstrip("_")
+    def __init__(self, store=None, delta=None, *args, **kwargs):
+        self._store = store or {}
+        self._delta = delta or {}
 
+    def __getattr__(self, key):
+        if self._delta and key in self._delta:
+            return self._delta[key]
 
-def _get(self, *args, **kwargs):
-    self._store = anvil.server.call(f"get_{_class_name(self)}", *args, **kwargs)
-    self._delta.clear()
+        return dict(self._store).get(key, None)
 
+    def __getitem__(self, key):
+        return getattr(self, key)
 
-def _add(self, *args, **kwargs):
-    self._store = anvil.server.call(
-        f"add_{_class_name(self)}", self._delta, *args, **kwargs
-    )
-    self._delta.clear()
+    def __setattr__(self, key, value):
+        is_private = key.startswith("_")
+        is_descriptor = hasattr(self.__class__, key) and hasattr(
+            getattr(self.__class__, key), "__set__"
+        )
+        if is_private or is_descriptor:
+            object.__setattr__(self, key, value)
+        else:
+            self._delta[key] = value
 
+    def __setitem__(self, key, value):
+        setattr(self, key, value)
 
-def _update(self, *args, **kwargs):
-    anvil.server.call(
-        f"update_{_class_name(self)}", self._store, self._delta, *args, **kwargs
-    )
-    self._delta.clear()
+    def _class_name(self):
+        return "".join(
+            "_" + c.lower() if c.isupper() else c for c in self.__class__.__name__
+        ).lstrip("_")
```

Comment:
might be better to cache this - maybe make it a property or even a class attribute

could be created inside decorator (or inside `__init_subclass__`)
(could even be a rare case of name mangling being of use (probably not))